src/core/event-handle.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Event Handle that is created by {@link EventHandler} and can be used for easier removing and events management.
+ * @example
+ * const evt = obj.on('test', (a, b) => {
+ *     console.log(a + b);
+ * });
+ * obj.fire('test');
+ *
+ * evt.off(); // easy way to remove this event
+ * obj.fire('test'); // this will not trigger an event
+ * @example
+ * // store an array of event handles
+ * let events = [ ];
+ *
+ * events.push(objA.on('testA', () => { }));
+ * events.push(objB.on('testB', () => { }));
+ *
+ * // when needed, remove all events
+ * events.forEach((evt) => {
+ *     evt.off();
+ * });
+ * events = [ ];
+ */
+class EventHandle {
+    /**
+     * @type {import('./event-handler.js').EventHandler}
+     * @private
+     */
+    handler;
+
+    /**
+     * @type {string}
+     * @private
+     */
+    name;
+
+    /**
+     * @type {import('./event-handler.js').HandleEventCallback}
+     * @private
+     */
+    callback;
+
+    /**
+     * @type {object}
+     * @private
+     */
+    scope;
+
+    /**
+     * @type {boolean}
+     * @private
+     */
+    once;
+
+    /**
+     * True if event has been removed.
+     * @type {boolean}
+     * @private
+     */
+    _removed = false;
+
+    /**
+     * @param {import('./event-handler.js').EventHandler} handler - source object of the event.
+     * @param {string} name - Name of the event.
+     * @param {import('./event-handler.js').HandleEventCallback} callback - Function that is called when event is fired.
+     * @param {object} scope - Object that is used as `this` when event is fired.
+     * @param {boolean} [once] - If this is a single event and will be removed after event is fired.
+     */
+    constructor(handler, name, callback, scope, once = false) {
+        this.handler = handler;
+        this.name = name;
+        this.callback = callback;
+        this.scope = scope;
+        this.once = once;
+    }
+
+    /**
+     * Remove references.
+     */
+    destroy() {
+        if (this._removed) return;
+        this._removed = true;
+    }
```

Comment:
Shouldn't a destroyed handle `off` itself at least?

Reply:
`destroy` is a private method of EventHandle, and cleans its data, and does not modify data outside of its class.

Reply:
Every other `destroy` method we have cleans up its events in other objects, usually in the form of:

```js
    destroy() {
        // remove any outstanding listeners
        this._registry.off("load", this._onLoad);
        this._registry.off("error", this._onError);
        // ...
    }
```

Why is this a special case?

Reply:
> Every other `destroy` method we have cleans up its events in other objects, usually in the form of:
> 
> ```js
>     destroy() {
>         // remove any outstanding listeners
>         this._registry.off("load", this._onLoad);
>         this._registry.off("error", this._onError);
>         // ...
>     }
> ```
> 
> Why is this a special case?
> 
> It also doesn't really do anything at all right now, is it a draft?

Events are "detached" rather than "destroyed". This PR does not change that and follows an existing pattern.
I've added clear references to destroy, like `this.callback = null` and other things, but tests did not like that. As the current implementation does not clear such references from event objects, their properties are accessed after during tests. So such change would have the potential to break.